include/ParameterSet.hpp

```diff
@@ -409,6 +411,79 @@ class ParameterSet {
     }
 
     /**
+     * Add a single GeographyPoint.
+     */
+    ParameterSet& addGeographyPoint(const GeographyPoint &val) {
+        validateType(WIRE_TYPE_GEOGRAPHY_POINT, false);
+        m_buffer.ensureRemaining(1 + 2*sizeof(double));
+        m_buffer.putInt8(WIRE_TYPE_GEOGRAPHY_POINT);
+        m_buffer.putDouble(val.getLongitude());
+        m_buffer.putDouble(val.getLatitude());
+        m_currentParam++;
+        return *this;
+    }
+
+    /**
+     * Add a vector of GeographyPoints.
+     */
+    ParameterSet& addGeographyPoint(const std::vector<GeographyPoint> &vals) {
+        validateType(WIRE_TYPE_GEOGRAPHY_POINT, true);
+        m_buffer.ensureRemaining(4 + static_cast<int32_t>(2*sizeof(double) * vals.size()));
+        m_buffer.putInt8(WIRE_TYPE_ARRAY);
+        m_buffer.putInt8(WIRE_TYPE_GEOGRAPHY_POINT);
+        m_buffer.putInt16(static_cast<int16_t>(vals.size()));
+        for (std::vector<GeographyPoint>::const_iterator i = vals.begin(); i != vals.end(); ++i) {
+            m_buffer.putDouble(i->getLongitude());
+            m_buffer.putDouble(i->getLatitude());
+        }
+        m_currentParam++;
+        return *this;
+    }
+
+    /**
+     * Add a single Geography.
+     */
+    ParameterSet& addGeography(const Geography &val) {
+        validateType(WIRE_TYPE_GEOGRAPHY, false);
+        // getSerializedSize() will return space for the
+        // size.
+        int32_t valSize = val.getSerializedSize();
+        m_buffer.ensureRemaining(1 + valSize);
+        // serializeTo will put the size in the buffer.
+        m_buffer.putInt8(WIRE_TYPE_GEOGRAPHY);
+        int realSize = val.serializeTo(m_buffer);
+        assert(valSize == realSize);
```

Comment:
This assertion is here in a debug build when I have made a mistake in calculating
the sizes.  These were hit all the time the first few times I did this, because I made
a lot of mistakes.  But after I corrected them things seem to work now.  They could be
removed, but they are only active in debug builds and it's possible, however unlikely, that
there are still errors.